Source/Engine/AGS.Engine/Graphics/Animations/AGSAnimationComponent.cs

```diff
@@ -50,7 +53,15 @@ public AnimationCompletedEventArgs Animate (IAnimation animation)
 		{
 			StartAnimation (animation);
 			return await animation.State.OnAnimationCompleted.Task;
-		}        
-	}
+		}
+
+        private void OnAnimationFrame()
+        {
+            // TODO: perhaps instead checking this every time, subscribe and unsubscribe
```

Comment:
I did not understand the todo comment: the on frame event (or rather, the property changed event as I mentioned above) is in animation state, part of the animation which is not a component: it's a parameter to the StartAnimation method, so there's no component add/remove for subscribing?

Reply:
What I meant here is that there possibly is a way to ensure this callback is only called if _spriteRender is not null:
* do not subscribe to the event unless sprite component is present;
* subscribe to the event when component is added (if animation is already running);
* unsubscribe from the event if component is removed.